src/Config_Command.php

```diff
@@ -153,16 +153,14 @@ public function create( $_, $assoc_args ) {
 			WP_CLI::error( '--dbprefix can only contain numbers, letters, and underscores.' );
 		}
 
-		$mysql_db_connection_args = [
-			'execute' => ';',
-			'host'    => $assoc_args['dbhost'],
-			'user'    => $assoc_args['dbuser'],
-			'pass'    => $assoc_args['dbpass'],
-		];
-
 		// Check DB connection.
 		if ( ! Utils\get_flag_value( $assoc_args, 'skip-check' ) ) {
-			Utils\run_mysql_command( '/usr/bin/env mysql --no-defaults', $mysql_db_connection_args );
+			// phpcs:disable WordPress.DB.RestrictedFunctions,WordPress.PHP.NoSilencedErrors.Discouraged
+			$mysql = mysqli_init();
+			if ( ! @mysqli_real_connect( $mysql, $assoc_args['dbhost'], $assoc_args['dbuser'], $assoc_args['dbpass'] ) ) {
+				die( 'Error (' . mysqli_connect_errno() . ') ' . mysqli_connect_error() );
```

Comment:
```suggestion
				die( 'Database connection error (' . mysqli_connect_errno() . ') ' . mysqli_connect_error() );
```

Reply:
I've read some caution against using `@` too liberally, and I think part of the reason is that error handlers still execute, and also that it can trap errors that we don't want to trap.

Not that it would demand any blocker on this PR, but it could bring some benefit to try and catch this in some other way. Potentially by setting `mysqli_report( MYSQLI_REPORT_STRICT )` we could wrap this in a `try` block and bypass generating warnings while also _only_ trapping errors related to the connection.

```php
mysqli_report( MYSQLI_REPORT_STRICT );
try {
	mysqli_real_connect( … );
} catch ( mysqli_sql_exception $e ) {
	die( "Database connection error ({$e->message})" );
}
```

Reply:
Thanks, I've adjusted the code based on the proposed approach.